src/renderer.js

```diff
@@ -109,7 +129,88 @@ document.addEventListener("DOMContentLoaded", async () => {
       console.error('Error handling group action via IPC:', e);
     }
   });
-  if (titleBar && tabBar) {
+
+  ipcRenderer.on('vertical-tabs-changed', async (_, enabled) => {
+    const oldBar = tabBar;
+    
+    // Clean up old webviews before creating new tab bar
+    if (oldBar && oldBar.webviews) {
+      oldBar.webviews.forEach((webview) => {
+        if (webview && webview.parentNode) {
+          webview.remove();
+        }
+      });
+      oldBar.webviews.clear();
+    }
+    
+    // Clear existing webviews from container
+    if (webviewContainer) {
+      while (webviewContainer.firstChild) {
+        webviewContainer.removeChild(webviewContainer.firstChild);
+      }
+    }
+    
+    // Remove old tab bar from DOM
+    if (oldBar && oldBar.parentElement) {
+      oldBar.remove();
+    }
+    
+    if (enabled) {
+      const { default: VerticalTabs } = await import('./pages/vertical-tabs.js');
+      tabBar = new VerticalTabs();
+      const keepExpanded = await ipcRenderer.invoke('settings-get', 'keepTabsExpanded');
+      if (keepExpanded) {
+        tabBar.updateKeepExpandedState(true);
+      }
+      document.body.appendChild(tabBar); // do NOT connect to titleBar
+    } else {
+      tabBar = new TabBar();
+      if (titleBar) {
+        titleBar.connectTabBar(tabBar);
+      } else {
+        document.body.appendChild(tabBar);
+      }
+    }
+    // Only connect AFTER append
+    tabBar.connectWebviewContainer(webviewContainer);
+    
+    // Force a layout update
+    setTimeout(() => {
+      if (tabBar.style) {
+        tabBar.style.display = 'none';
+        tabBar.offsetHeight; // Trigger reflow
+        tabBar.style.display = '';
+      }
```

Comment:
This is clearing and refreshing the horizontal tab title bar display, but technically it’s not "removing" it. The horizontal bar should completely collapse when vertical tabs are enabled.

<img width="798" height="91" alt="Screenshot 2025-08-11 at 10 00 54 AM" src="https://github.com/user-attachments/assets/34019d92-d14a-4ec4-bdbd-4e0dfa0b3abc" />

Reply:
Also, on vertical tabs mode, I cannot drag the window from nav bar.

<img width="859" height="421" alt="Screenshot 2025-08-11 at 11 12 07 AM" src="https://github.com/user-attachments/assets/dadccc4b-4d95-4270-9edb-2db8633353b5" />

Reply:
To make it draggable, you can simply do this:

`-webkit-app-region: drag;`